Firestore/core/src/remote/datastore.cc

```diff
@@ -108,6 +108,7 @@ Datastore::Datastore(
       connectivity_monitor_{connectivity_monitor},
       grpc_connection_{database_info, worker_queue, &grpc_queue_,
                        connectivity_monitor_, firebase_metadata_provider},
+      database_info_{database_info},
```

Comment:
save the parameter database_info as class attribute to read datastore.databaseId out of it later.